qualtran/bloqs/arithmetic/bitwise.py

```diff
@@ -118,3 +122,64 @@ def _cxork() -> XorK:
     cxork = XorK(QUInt(8), 0b01010111).controlled()
     assert isinstance(cxork, XorK)
     return cxork
+
+
+@frozen
+class Xor(Bloq):
```

Comment:
Can we follow the same signature as `XorK` for consistency? This would mean:

a) Supporting singly controlled version by deriving from `SpecializedSingleQubitControlledGate` and have an attribute `control_val: Optional[int] = None`.
b) Use `dtype: QDType` instead of `bitsize: SymbolicInt`
c) Rename registers to `x` and `y` s.t. you xor `x` into the `y` register. This is mainly so the controlled version has. a register called `ctrl`. We also often use `x` / `y` or `a`/`b` in arithmetic bloqs to denote source / target registers. 

Among the above 3 suggestions, I think maybe (a) is redundant if we just fallback to using `Controlled(Xor())`, but (b) and (c) should be incorporated irrespective of consistency with `XorK`.